src/stravalib/protocol.py

```diff
@@ -131,18 +169,19 @@ def _request(
             The request method (GET/POST/etc.)
         check_for_errors : bool
             Whether to raise an error or not.
-        refresh_expired_token : bool (default True)
-            Whether to automagically refresh the users token or not.
-            Requires environment setup to work.
 
         Returns
         -------
         Dict[str, Any]
             The parsed JSON response.
         """
-        # Try to refresh the token unless told to not try
-        if refresh_expired_token:
-            self.refresh_expired_token()
+
+        # Check if refresh token credentials are setup
+        credentials = self._check_credentials()
```

Comment:
It's great that you refactored this 👍

However, it may be better to move this check to `APIV3`s initializer. Otherwise, this check (and its associated warning/error messages) will be performed on every request.

Reply:
Ok this has been moved to init. it's much better now. I'm really glad that you caught that. I'm refactoring tests a bit and I'll look at that comment above.